### PR TITLE
Bump golang to 1.19.4-alpine3.17, rancher/wharfie to v0.5.3, and dapper to v0.6.0

### PR DESCRIPTION
**Pull request by Oats87**
https://github.com/rancher/system-agent/issues/99

### CODE REVIEW

**Comment by Oats87** [comment]
Validated that `rancher-system-agent` is still able to pull images from a private registry, see truncated logs:
```
Dec 28 22:31:38 ck-c7-5 rancher-system-agent[1888]: time="2022-12-28T22:31:38-08:00" level=info msg="Using private registry config file at /etc/rancher/agent/registries.yaml"
Dec 28 22:31:38 ck-c7-5 rancher-system-agent[1888]: time="2022-12-28T22:31:38-08:00" level=info msg="Pulling image izdev1-registry.qa.rancher.space/rancher/system-agent-installer-rke2:v1.24.7-rke2r1"
Dec 28 22:31:43 ck-c7-5 rancher-system-agent[1888]: time="2022-12-28T22:31:43-08:00" level=info msg="Extracting file installer.sh to /var/lib/rancher/agent/work/20221228-223138/c9d3b3af9286a67c019b3eb6a5af8fcc508148687cb49020c518bd347094880e_0/installer.sh"
Dec 28 22:31:43 ck-c7-5 rancher-system-agent[1888]: time="2022-12-28T22:31:43-08:00" level=info msg="Extracting file rke2.linux-amd64.tar.gz to /var/lib/rancher/agent/work/20221228-223138/c9d3b3af9286a67c019b3eb6a5af8fcc508148687cb49020c518bd347094880e_0/rke2.linux-amd64.tar.gz"
Dec 28 22:31:48 ck-c7-5 rancher-system-agent[1888]: time="2022-12-28T22:31:48-08:00" level=info msg="Extracting file sha256sum-amd64.txt to /var/lib/rancher/agent/work/20221228-223138/c9d3b3af9286a67c019b3eb6a5af8fcc508148687cb49020c518bd347094880e_0/sha256sum-amd64.txt"
Dec 28 22:31:48 ck-c7-5 rancher-system-agent[1888]: time="2022-12-28T22:31:48-08:00" level=info msg="Extracting file run.sh to /var/lib/rancher/agent/work/20221228-223138/c9d3b3af9286a67c019b3eb6a5af8fcc508148687cb49020c518bd347094880e_0/run.sh"
Dec 28 22:31:48 ck-c7-5 rancher-system-agent[1888]: time="2022-12-28T22:31:48-08:00" level=info msg="[Applyinator] Running command: sh [-c run.sh]"
```

**Comment by Oats87** [comment]
@jakefhyde @snasovich This PR is now ready for review -- CI runs for the `rancher/system-agent` repository again, and the custom built rancher-system-agent using these changes is still functional with a private registry as shown in my comment above.

File: pkg/image/util.go [hunk 1]
```diff
@@ -90,29 +90,28 @@ func (u *Utility) Stage(destDir string, imgString string) error {
...
 				return err
 			}
-			kcs = append(kcs, plugins)
+			registry.DefaultKeychain = plugins
 		} else {
-			kcs = append(kcs, authn.DefaultKeychain)
+			// The kubelet image credential provider plugin also falls back to checking legacy Docker credentials, so only
+			// explicitly set up the go-containerregistry DefaultKeychain if plugins are not configured.
+			// DefaultKeychain tries to read config from the home dir, and will error if HOME isn't set, so also gate on that.
+			if os.Getenv("HOME") != "" {
```
Comment (snasovich) [inline pkg/image/util.go:108]: Given the explanation above, maybe it's a good idea to at least log a warning if HOME is not set as we currently just silently not setting `DefaultKeychain`.

**Review by snasovich — APPROVED** [review]
LGTM, one suggestion/question

File: pkg/image/util.go (hunk 1 above)
Comment (Oats87) [reply pkg/image/util.go:108]: Our desired use case does not have a user directly interfacing with the system-agent, so I am not too "log happy" with it from an interactive perspective

**Review by jakefhyde — APPROVED** [review]

**Comment by pgonin** [comment]
Any plan to switch from alpine to BCI ?

**Comment by Oats87** [comment]
> Any plan to switch from alpine to BCI ?

Not currently, but I would like to officially schedule that instead of just ad-hoc making the change.

**Pull request merged**